mission_control/navigator_missions/navigator_singleton/pose_editor.py

```diff
@@ -0,0 +1,201 @@
+from __future__ import division
+import warnings
+
+import numpy as np
+from tf import transformations
+from nav_msgs.msg import Odometry
+from uf_common.msg import PoseTwistStamped, PoseTwist, MoveToGoal
+from geometry_msgs.msg import Pose, PoseStamped, Quaternion, Point, Vector3, Twist
+from sub8_ros_tools import rosmsg_to_numpy, make_header
+
+
+UP = np.array([0.0, 0.0, 1.0], np.float64)
+EAST, NORTH, WEST, SOUTH = [transformations.quaternion_about_axis(np.pi / 2 * i, UP) for i in xrange(4)]
+UNITS = {'m': 1, 'ft': 0.3048, 'yard': 0.9144, 'rad': 1, 'deg': 0.0174533}
+
+def normalized(x):
+    x = np.array(x)
+    if max(map(abs, x)) == 0:
+        warnings.warn('Normalizing zero-length vector to random unit vector')
```

Comment:
Don't these kind of errors ocur in several places? It would be nice to include the file name or node name in the message.